src/types.ts

```diff
@@ -56,6 +56,16 @@ export interface IPackageJSONService {
    * Writes the JSON to the specified file.
    */
   writePackageFile(filePath: string, fileContents: IPackageFile): Promise<void>
+
+  /**
+   * Reads and parses YAML from the specified file. Path is relative to the current working directory.
+   */
+  readPnpmWorkspaceFile(
```

Comment:
Yeah I agree that putting this here isn't ideal. As mentioned elsewhere, maybe we need a workspaces resolver abstraction that can be implemented for each package manager.